keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -493,20 +496,31 @@ More information on Windows resource access can be found at https://docs.microso
 Note: there will be no `chroot` equivalent.
 - An environment variable `$CONTAINER_SANDBOX_MOUNT_POINT` will be set to the absolute path where the container volume is mounted.
 - Volume mounts (including service account tokens) will be supported for privileged containers and will be mounted under the container volume. Programs running inside the container can either access volume mounts be using a relative path or by prefixing `$CONTAINER_SANDBOX_MOUNT_POINT` to their paths (example: use either `.\var\run\secrets\kubernetes.io\serviceaccount\` or `$CONTAINER_SANDBOX_MOUNT_POINT\var\run\secrets\kubernetes.io\serviceaccount\` to access service account tokens). These relative paths will be based on `Pod.containers.volumeMounts.mountPath`.
-- Client libraries such as https://pkg.go.dev/k8s.io/client-go/rest#InClusterConfig may be updated to prefix paths with `$CONTAINER_SANDBOX_MOUNT_POINT` if the environment variable is set for Windows so these libraries will work in `hostProcess`containers. This will be re-evaluated when transistioning from `alpha` to `beta` as we get more feedback.
-Note: it is not possible to feature-gate this behavior in client libraries and because of this the functionality should not be added to client libraries after privileged containers while this feature is in `alpha`.
-- Named Pipe mounts will **not** be supported. Instead named pipes should access via their path on the host (\\\\.\\pipe\\*). Unix domain sockets mounts **will** be supported.
-- All other volume types supported for normal containers on Windows will work with privileged containers.
+  - Note: We are prototyping a new approach to how the file system is created for `hostProcess` containers that would present the filesystem in a similar manner to non-hostProcess containers running on Windows (`c:\` would be the root instead of `c:\c\<container id>`).
+  This would make it so files from volume mounts would be accessible via static paths. HostProcess containers would still have full access to the host file-system.
+  https://github.com/microsoft/hcsshim/pull/1107 is tracking this exploratory work.
+  This functionality will most-likely not be ready during Kubernetes v1.23 and any changes made to how volume mounts work would be done before this features becomes stable.
```

Comment:
is it a good idea to proceed with an alternate-root envvar approach that has to propagate into all containerized programs if it's possible this will make that unnecessary in the long term?

Reply:
@jsturtevant and others debated this at length.
Progressing Cluster API support for Windows  (and a few other highly requested scenarios) is blocked by not having HostProcess container support.
We were thinking that the sooner we move HostProcess containers to beta the sooner we can start collecting feedback around real usage scenarios.

Reply:
Also the alternate-root envvar will be propagated even if we change how volume mounts so containerized applications built around the current behavior won't need to be updated if/when we do change behaviors.

Reply:
> Also the alternate-root envvar will be propagated even if we change how volume mounts are configured so containerized applications built around the current behavior won't need to be updated if/when we do change behaviors.

Yep. If/when we do make any changes, we'd set the envvar to "C:\" to mimic the behavior we have currently where the envvar refers to the root of the containers volume and make it so anyone that had used this previously shouldn't have to change anything.

Reply:
will the envvar be usable in the pod manifest as `$(CONTAINER_SANDBOX_MOUNT_POINT)` so it can be used in command line args?

can you add details about how it should be consumed in the pod manifest and in the container process in a way that works whether the value is set or not? specifically for the following types of pods:

* normal (non-hostProcess) pods
* hostProcess pods (currently, which have to use $CONTAINER_SANDBOX_MOUNT_POINT)
* hostProcess pods (in the theoretical future, when $CONTAINER_SANDBOX_MOUNT_POINT is not necessary)

will the value have a trailing slash? will it be forward or backward-slash delimited?

Reply:
Yea, let me update the KEP with this information.

Reply:
I updated the KEP with some of the information you provided.
All slashes will be backward-slashes (Windows paths always use \'s for paths but some shells like powershell know how to handle /'s. 
(fun fact: this is because historically windows programs used forward slashes for command line switches / arguments and a lot of those older programs do a string split on '/'. This would cause problems if paths were used as arguments so Windows just sticks with \ in paths)

We are including a trailing slash (which is called out) because it is common practice for windows paths to always end in a slash. Windows APIs can handle paths with multiple slashes (like c:\\foo\ ) so even if path joining calls aren't smart enough to check for a trailing \ in the first path windows this mostly work as expected.

Reply:
I added some links to working manifests that configure calico CNI on on Windows nodes using hostProcess containers too!

Reply:
Also, we don't expect workloads that are required to run in hostProcess containers to run in normal pods.
For the current behavior we expect that all hostProcess containers that need to access files inside volume mounts to use $CONTAINER_SANDBOX_MOUNT_POINT. 

@liggitt does this address your questions?

Reply:
Also the doc was updated.